src/vlrs.jl

```diff
@@ -124,33 +124,69 @@ The registered record IDs associated to VLRs of this record type `TData`
 """
 official_record_ids(::Type{TData}) where TData = error("Official record IDs not set for VLRs with data type $(TData)")
 
+"""
+    $(TYPEDSIGNATURES)
+
+Get the data type associated with a particular `user_id` and `record_id`. 
+This is used to automatically parse VLR data types on reading
+"""
+function data_type_from_ids(user_id::String, record_id::Integer)
+    registered_vlr_types = get_all_vlr_types()
+
+    for type ∈ registered_vlr_types
+        if (user_id == official_user_id(type)) && (record_id ∈ official_record_ids(type))
+            return type
+        end
+    end
+
+    error("Can't find VLR data type to parse for user ID $(user_id) and record ID $(record_id)")
+end
+
+function get_all_vlr_types()
+    # eval to make sure we get the methods in the global scope - use both the user and record IDs methods to check for types
+    ms = @eval vcat(methods(LASDatasets.official_user_id), methods(LASDatasets.official_record_ids))
```

Comment:
you may get away with Base.invokelatest here rather than @eval?

Reply:
I think `Base.invokelatest` needs you to call the function explicitly rather than just get the methods?

Reply:
yeah would only be if got the Val thing working